e2e/tests/session.feature

```diff
@@ -8,3 +8,8 @@ Feature: Session Namespace
 #  TODO_IMPROVE: ...
 #    And an event should be emitted...
 #  TODO_INCOMPLETE: add step(s) for proof validation.
+
+#  Scenario: Late RelayRequests are handled within the session grace period
+#    When a RelayRequest is signed for an Application within session number N
+#    And the RelayMiner gets the RelayRequest within session number N+1
+#    Then the RelayMiner should process the RelayRequest and include it in the claim N
```

Comment:
Wdyt of something like this instead
```suggestion
# TODO_BLOCKER(@red-0ne): Make sure to implement and validate this test
Scenario: A late Relay inside the SessionGracePeriod is handled
    Given the user has the pocketd binary installed
    And the parameter "NumBlockPerSession" is "4"
    And the parameter "SessionGracePeriod" is "1"
    When the supplier "supplier1" has serviced a session with service "svc1" for application "app1" with session number "1"
    And we have waited "5" block
    And the application "app1" sends a relay request to supplier "supplier1" for service "svc" with session number "1"
    Then the application "app1" receives a successful relay response signed by "supplier1" for session number "1"
    And after the supplier "supplier1" updates a claim for session number "1" for service "svc1" for application "app1"
    Then the claim created by supplier "supplier1" for service "svc1" for application "app1" should be persisted on-chain

# TODO_BLOCKER(@red-0ne): Make sure to implement and validate this test
Scenario: A late Relay outside the SessionGracePeriod is rejected
    Given the user has the pocketd binary installed
    And the parameter "NumBlockPerSession" is "4"
    And the parameter "SessionGracePeriod" is "1"
    When the supplier "supplier1" has serviced a session with service "svc1" for application "app1" with session number "1"
    And we have waited "10" block
    And the application "app1" sends a relay request to supplier "supplier1" for service "svc" with session number "1"
    Then the application "app1" receives a failed relay response
    And the supplier "supplier1" fails to update a claim for session number "1" for service "svc1" for application "app1"
```

Reply:
@Olshansk, Merged these as they are more explicit but I don't think they will exercise the session rollover.

First it has to start with the application sending a relay in a block height that resolves to session number `1`
Then blocks has to pass (wait n blocks) so when the supplier receives the relay from the application, the supplier is already in a block that resolves to session number `2`.

The nuance is that the blocks has to pass between the application sending the relay request and the supplier receiving that relay request.

Reply:
> The nuance is that the blocks has to pass between the application sending the relay request and the supplier receiving that relay request.

Can you update the steps to reflect that? Can we not do something like:

1. The supplier services a relay for session number 1
2. The supplier waits 5 blocks
3. The supplier calls GetSession and gets session number 2
4. The supplier services a relay session number 1

I think that if we get really explicit with the waiting and the numbers, we can make it deterministic, which is the goal of the test. What do you think we're missing here?

Reply:
@Olshansk , I updated the test, WDYT of of it's stated now?